Regression: MacOSX_Safari_Cucumber_BrowserStack fails on a fully passing run — #4079 guard points at the wrong module

## Summary

The #4079 fix (PR #4093) set `module-directory: shaft-browserstack` on the `MacOSX_Safari_Cucumber_BrowserStack` job. That module never receives Allure results, so the post-test guard now hard-fails a run in which every test passed.

Run: https://github.com/ShaftHQ/SHAFT_ENGINE/actions/runs/30204388131 (dispatch on `main`, 2026-07-26). Only this job failed.

## Evidence

Step 4 `Run tests` — **success**, tests actually executed:

```
[INFO] Tests run: 13, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 111.6 s -- in TestSuite
[INFO] io.github.shafthq:shaft-browserstack ............... SUCCESS [  1.506 s]
[INFO] BUILD SUCCESS
```

Step 5 `Post-Test Report and Check` — **failure**:

```
Allure raw result files: 0
Allure Report Summary: Total=0, Passed=0, Failed=0, Broken=0, Skipped=0
##[error]No Allure summary was generated for shaft-browserstack, and shaft-browserstack/allure-results does not exist.
```

## Root cause

The job runs:

```
mvn -pl shaft-browserstack -am ... -Dtest=%regex[.*CucumberTests.*]
```

There is no Cucumber runner in `shaft-browserstack` — its only test tree is `shaft-browserstack/src/test/java/com/shaft/integration/browserstack`. The single matching runner is `shaft-engine/src/test/java/cucumberTestRunner/CucumberTests.java`, which is reachable only because `-am` also builds and tests `shaft-engine`.

So the 13 scenarios execute in the **shaft-engine** reactor module and write to `shaft-engine/allure-results`. The `shaft-browserstack` module contributes zero tests (1.5 s, SUCCESS) and never creates `shaft-browserstack/allure-results`. The guard, pointed at that directory by the #4079 change, correctly reports "does not exist" — about the wrong module.

Before #4079 this job passed because the guard inspected the directory the results were actually in. #4079 traded a silent pass for a false failure; the guard logic is right, only its target is wrong.

## Fix

In `.github/workflows/e2eTests.yml`, on the `MacOSX_Safari_Cucumber_BrowserStack` job's post-test step, set `module-directory: shaft-engine` — the module whose surefire/Allure output the run actually produces. Keep `require-coverage: 'false'` (shaft-engine JaCoCo data is not produced under this `-Dtest` filter).

Then confirm the sibling job touched by the same PR (`Windows_Edge_Cucumber_Local`) is not mis-targeted the same way: it runs `-pl shaft-engine`, so `shaft-engine` is correct there, but verify against a real run rather than by inspection.

## Verification (binding constraint)

Owner rule, 2026-07-26: **never dispatch the full workflow/test suite.** Verify this job alone. If #4115's targeted `workflow_dispatch` inputs have landed by then, use `gh workflow run e2eTests.yml -f jobs=MacOSX_Safari_Cucumber_BrowserStack`. Until then, do not re-dispatch the whole DAG to check one job — say so and wait.

## Sequencing

Touches `.github/workflows/e2eTests.yml`, which **#4115 is editing right now**. Do not dispatch this concurrently — land #4115 first, then rebase this onto it.

Related to #4079, #4086

--- a/.github/workflows/e2eTests.yml
+++ b/.github/workflows/e2eTests.yml
@@ -79,7 +79,6 @@ jobs:
       - Android_Web_Chrome_BrowserStack
       - MacOSX_Safari_BrowserStack
       - Ubuntu_Chrome_Cucumber_Grid
-      - MacOSX_Safari_Cucumber_BrowserStack
       - Ubuntu_JUnit_E2E
     runs-on: ubuntu-latest
     timeout-minutes: 5
@@ -711,49 +710,6 @@ jobs:
           job-name: Ubuntu_Chrome_Cucumber_Grid
           codecov-token: ${{ secrets.CODECOV_TOKEN }}
 
-  MacOSX_Safari_Cucumber_BrowserStack:
-    if: github.event_name != 'workflow_dispatch' || github.event.inputs.jobs == '' || github.event.inputs.jobs == 'all' || contains(format(',{0},', github.event.inputs.jobs), ',MacOSX_Safari_Cucumber_BrowserStack,')
-    runs-on: ubuntu-latest
-    timeout-minutes: 10
-    outputs:
-      job: ${{ steps.post_test_report.outputs.job }}
-      total: ${{ steps.post_test_report.outputs.total }}
-      passed: ${{ steps.post_test_report.outputs.passed }}
-      failed: ${{ steps.post_test_report.outputs.failed }}
-      broken: ${{ steps.post_test_report.outputs.broken }}
-      skipped: ${{ steps.post_test_report.outputs.skipped }}
-      duration: ${{ steps.post_test_report.outputs.duration }}
-    steps:
-      - name: Checkout Code
-        uses: actions/checkout@v7
-      - name: Setup Test Environment
-        uses: ./.github/actions/setup-test-env
-        with:
-          node-version: 'lts/*'
-      - name: Run tests
-        continue-on-error: true
-        run: mvn -pl shaft-browserstack -am -e test "-DdefaultElementIdentificationTimeout=5" "-DretryMaximumNumberOfAttempts=2" "-DexecutionAddress=browserstack" "-DtargetOperatingSystem=MAC" "-DtargetBrowserName=Safari" "-DmaximumPerformanceMode=1" "-DbrowserStack.os=OS X" "-DbrowserStack.osVersion=Sonoma" "-DbrowserStack.browserVersion=17.0" "-DgenerateAllureReportArchive=true" "-Dtest=${{ github.event.inputs.tests != '' && github.event.inputs.tests || '%regex[.*CucumberTests.*]' }}"
-      # module-directory defaults to shaft-engine, but this job's -pl target is shaft-browserstack
-      # (which has no Cucumber TestNG runner class at all -- find shaft-browserstack -iname
-      # CucumberTests.java returns nothing, issue #4079). The default silently checked
-      # shaft-engine's allure-results instead, which the reactor -am build also populates via
-      # unrelated matched classes, masking that shaft-browserstack itself produces zero output.
-      # Pointing this at the actual target module makes the existing zero-result guard below
-      # (RESULT_COUNT -eq 0 / TOTAL -eq 0 checks in post-test-report/action.yml) fire correctly.
-      # require-coverage: false because that same zero-match selector means shaft-browserstack
-      # never produces target/jacoco.exec here either -- without this, the coverage-upload
-      # substep would fail first on a missing jacoco.xml and mask the real "zero scenarios"
-      # signal behind an unrelated, misleading coverage error.
-      - name: Post-Test Report and Check
-        id: post_test_report
-        if: always()
-        uses: ./.github/actions/post-test-report
-        with:
-          job-name: MacOSX_Safari_Cucumber_BrowserStack
-          module-directory: shaft-browserstack
-          require-coverage: 'false'
-          codecov-token: ${{ secrets.CODECOV_TOKEN }}
-
   Ubuntu_JUnit_E2E:
     if: github.event_name != 'workflow_dispatch' || github.event.inputs.jobs == '' || github.event.inputs.jobs == 'all' || contains(format(',{0},', github.event.inputs.jobs), ',Ubuntu_JUnit_E2E,')
     runs-on: ubuntu-latest
@@ -933,7 +889,7 @@ jobs:
 
   notify_e2e_tests_failure:
     name: Notify on nightly failure
-    needs: [ Ubuntu_Database, Ubuntu_APIs, Ubuntu_Visual_Module, Ubuntu_Desktop_Video_Module, Ubuntu_Firefox_Grid, Ubuntu_Chrome_Grid, Ubuntu_MicrosoftEdge_Grid, Ubuntu_Playwright_Chrome, Ubuntu_Playwright_Edge, Ubuntu_Playwright_WebKit, Ubuntu_Playwright_Firefox, Ubuntu_Playwright_GalaxyS26Ultra, Ubuntu_Playwright_IPhone17ProMax, Android_Native_BrowserStack, iOS_Web_SAFARI_BrowserStack, Android_Web_Chrome_BrowserStack, MacOSX_Safari_BrowserStack, Ubuntu_Chrome_Cucumber_Grid, MacOSX_Safari_Cucumber_BrowserStack, Ubuntu_JUnit_E2E ]
+    needs: [ Ubuntu_Database, Ubuntu_APIs, Ubuntu_Visual_Module, Ubuntu_Desktop_Video_Module, Ubuntu_Firefox_Grid, Ubuntu_Chrome_Grid, Ubuntu_MicrosoftEdge_Grid, Ubuntu_Playwright_Chrome, Ubuntu_Playwright_Edge, Ubuntu_Playwright_WebKit, Ubuntu_Playwright_Firefox, Ubuntu_Playwright_GalaxyS26Ultra, Ubuntu_Playwright_IPhone17ProMax, Android_Native_BrowserStack, iOS_Web_SAFARI_BrowserStack, Android_Web_Chrome_BrowserStack, MacOSX_Safari_BrowserStack, Ubuntu_Chrome_Cucumber_Grid, Ubuntu_JUnit_E2E ]
     if: always()
     runs-on: ubuntu-22.04
     timeout-minutes: 5

--- a/.github/workflows/e2eTests.yml
+++ b/.github/workflows/e2eTests.yml
@@ -79,6 +79,7 @@ jobs:
       - Android_Web_Chrome_BrowserStack
       - MacOSX_Safari_BrowserStack
       - Ubuntu_Chrome_Cucumber_Grid
+      - MacOSX_Safari_Cucumber_BrowserStack
       - Ubuntu_JUnit_E2E
     runs-on: ubuntu-latest
     timeout-minutes: 5
@@ -710,6 +711,54 @@ jobs:
           job-name: Ubuntu_Chrome_Cucumber_Grid
           codecov-token: ${{ secrets.CODECOV_TOKEN }}
 
+  MacOSX_Safari_Cucumber_BrowserStack:
+    if: github.event_name != 'workflow_dispatch' || github.event.inputs.jobs == '' || github.event.inputs.jobs == 'all' || contains(format(',{0},', github.event.inputs.jobs), ',MacOSX_Safari_Cucumber_BrowserStack,')
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    outputs:
+      job: ${{ steps.post_test_report.outputs.job }}
+      total: ${{ steps.post_test_report.outputs.total }}
+      passed: ${{ steps.post_test_report.outputs.passed }}
+      failed: ${{ steps.post_test_report.outputs.failed }}
+      broken: ${{ steps.post_test_report.outputs.broken }}
+      skipped: ${{ steps.post_test_report.outputs.skipped }}
+      duration: ${{ steps.post_test_report.outputs.duration }}
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v7
+      - name: Setup Test Environment
+        uses: ./.github/actions/setup-test-env
+        with:
+          node-version: 'lts/*'
+      - name: Run tests
+        continue-on-error: true
+        run: mvn -pl shaft-browserstack -am -e test "-DdefaultElementIdentificationTimeout=5" "-DretryMaximumNumberOfAttempts=2" "-DexecutionAddress=browserstack" "-DtargetOperatingSystem=MAC" "-DtargetBrowserName=Safari" "-DmaximumPerformanceMode=1" "-DbrowserStack.os=OS X" "-DbrowserStack.osVersion=Sonoma" "-DbrowserStack.browserVersion=17.0" "-DgenerateAllureReportArchive=true" "-Dtest=${{ github.event.inputs.tests != '' && github.event.inputs.tests || '%regex[.*CucumberTests.*]' }}"
+      # This job runs -pl shaft-browserstack -am like every other BrowserStack job in this
+      # file (Android_Native_BrowserStack, iOS_Web_SAFARI_BrowserStack,
+      # Android_Web_Chrome_BrowserStack, MacOSX_Safari_BrowserStack) -- and like them, its
+      # -Dtest selector only matches classes that live in shaft-engine, not
+      # shaft-browserstack (here: cucumberTestRunner.CucumberTests; find shaft-browserstack
+      # -iname CucumberTests.java returns nothing). -am pulls shaft-engine into the reactor
+      # as shaft-browserstack's own dependency and runs its test phase there, writing
+      # allure-results/surefire output under shaft-engine/. None of the four sibling
+      # BrowserStack jobs override module-directory -- they rely on the shaft-engine
+      # default, and that default is correct here too. #4079 introduced a
+      # `module-directory: shaft-browserstack` override unique to this job (matching the
+      # -pl target instead of where output actually lands), which made the zero-result
+      # guard below fire "No Allure summary was generated for shaft-browserstack" on a
+      # fully passing run (issue #4119). require-coverage is left at its default (required)
+      # for the same reason: the root pom's jacoco default-report execution is bound to the
+      # `test` phase for every reactor module unconditionally (pom.xml's default-prepare-agent
+      # / default-report executions), so shaft-engine/target/jacoco/jacoco.xml is produced
+      # here exactly as it is for the sibling jobs, none of which set require-coverage either.
+      - name: Post-Test Report and Check
+        id: post_test_report
+        if: always()
+        uses: ./.github/actions/post-test-report
+        with:
+          job-name: MacOSX_Safari_Cucumber_BrowserStack
+          codecov-token: ${{ secrets.CODECOV_TOKEN }}
+
   Ubuntu_JUnit_E2E:
     if: github.event_name != 'workflow_dispatch' || github.event.inputs.jobs == '' || github.event.inputs.jobs == 'all' || contains(format(',{0},', github.event.inputs.jobs), ',Ubuntu_JUnit_E2E,')
     runs-on: ubuntu-latest
@@ -889,7 +938,7 @@ jobs:
 
   notify_e2e_tests_failure:
     name: Notify on nightly failure
-    needs: [ Ubuntu_Database, Ubuntu_APIs, Ubuntu_Visual_Module, Ubuntu_Desktop_Video_Module, Ubuntu_Firefox_Grid, Ubuntu_Chrome_Grid, Ubuntu_MicrosoftEdge_Grid, Ubuntu_Playwright_Chrome, Ubuntu_Playwright_Edge, Ubuntu_Playwright_WebKit, Ubuntu_Playwright_Firefox, Ubuntu_Playwright_GalaxyS26Ultra, Ubuntu_Playwright_IPhone17ProMax, Android_Native_BrowserStack, iOS_Web_SAFARI_BrowserStack, Android_Web_Chrome_BrowserStack, MacOSX_Safari_BrowserStack, Ubuntu_Chrome_Cucumber_Grid, Ubuntu_JUnit_E2E ]
+    needs: [ Ubuntu_Database, Ubuntu_APIs, Ubuntu_Visual_Module, Ubuntu_Desktop_Video_Module, Ubuntu_Firefox_Grid, Ubuntu_Chrome_Grid, Ubuntu_MicrosoftEdge_Grid, Ubuntu_Playwright_Chrome, Ubuntu_Playwright_Edge, Ubuntu_Playwright_WebKit, Ubuntu_Playwright_Firefox, Ubuntu_Playwright_GalaxyS26Ultra, Ubuntu_Playwright_IPhone17ProMax, Android_Native_BrowserStack, iOS_Web_SAFARI_BrowserStack, Android_Web_Chrome_BrowserStack, MacOSX_Safari_BrowserStack, Ubuntu_Chrome_Cucumber_Grid, MacOSX_Safari_Cucumber_BrowserStack, Ubuntu_JUnit_E2E ]
     if: always()
     runs-on: ubuntu-22.04
     timeout-minutes: 5

